face/EnrollAndRecognizePersons/python/recognize_persons.ipynb

```diff
@@ -27,6 +27,50 @@
    "source": [
     "import os\n",
     "from shared_functions import identify_faces, delete_dynamic_person_group, delete_person\n",
+    "from PIL import Image, ImageDraw, ImageFont\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "def plot_faces_on_image(image_path, identified_faces):\n",
+    "    # Load the original image\n",
+    "    image = Image.open(image_path)\n",
+    "    draw = ImageDraw.Draw(image)\n",
+    "    \n",
+    "    # Load a font\n",
+    "    try:\n",
+    "        font = ImageFont.truetype(\"arial.ttf\", 20)\n",
+    "    except IOError:\n",
+    "        font = ImageFont.load_default()\n",
+    "\n",
+    "    # Iterate over the identified faces and draw bounding boxes and names\n",
+    "    for face in identified_faces:\n",
+    "        bbox = face['bbox']\n",
+    "        name = face['personName'] if face['personName'] else \"Unknown\"\n",
+    "        \n",
+    "        # Define the bounding box coordinates\n",
+    "        left = bbox['left']\n",
+    "        top = bbox['top']\n",
+    "        right = left + bbox['width']\n",
+    "        bottom = top + bbox['height']\n",
+    "        \n",
+    "        # Draw the bounding box\n",
+    "        draw.rectangle([(left, top), (right, bottom)], outline=\"green\", width=2)\n",
+    "        \n",
+    "        # Calculate text size using textbbox\n",
+    "        text_bbox = draw.textbbox((0, 0), name, font=font)\n",
+    "        text_width = text_bbox[2] - text_bbox[0]\n",
+    "        text_height = text_bbox[3] - text_bbox[1]\n",
+    "        \n",
+    "        # Draw a rectangle behind the text\n",
+    "        draw.rectangle([(left, top - text_height - 4), (left + text_width + 4, top)], fill=\"grey\")\n",
```

Comment:
Is it possible that this is an edge-of-field-of-view face so (top - text_height - 4) becomes negative?